### ISSUE TITLE
IconScraper throws a lot errors

### ISSUE THREAD

**Issue by RainerEmrich** [nextcloud/mail#647]
### Steps to reproduce
1. Update mail to v0.7.5
2. Start mail app
3. Inspect Log

### Expected behaviour
No Errors should be thrown.
### Actual behaviour
A lot  errors are thrown, see log below (excerpt).

### Mail app

**Mail app version:** v0.7.5

**Mailserver or service:** 1&1 Imap Service

**Transport security - IMAP:** SSL/TLS

**Transport security - SMTP:** STARTTLS

**Number of accounts:** 2

## Server configuration detail

**Operating system:** Linux 4.4.0-101-generic #124-Ubuntu SMP Fri Nov 10 18:29:59 UTC 2017 x86_64

**Webserver:** Apache (fpm-fcgi)

**Database:** mysql 10.2.11

**PHP version:** 7.1.12-1+ubuntu16.04.1+deb.sury.org+1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, cgi-fcgi, mysqlnd, PDO, xml, apcu, apc, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imagick, imap, intl, json, ldap, exif, mcrypt, msgpack, mysqli, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, readline, redis, shmop, SimpleXML, smbclient, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, memcached, libsmbclient, Zend OPcache

**Nextcloud version:** 12.0.4 RC2 - 12.0.4.1

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - activity: 2.5.2
 - admin_audit: 1.2.0
 - admin_notifications: 1.0.1
 - announcementcenter: 3.1.1
 - apporder: 0.4.0
 - audioplayer: 2.2.1
 - bruteforcesettings: 1.0.2
 - calendar: 1.5.6
 - caniupdate: 0.1.0
 - checksum: 0.3.5
 - circles: 0.13.6
 - comments: 1.2.0
 - contacts: 2.0.1
 - dav: 1.3.0
 - deck: 0.2.8
 - external: 2.0.3
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_accesscontrol: 1.2.5
 - files_automatedtagging: 1.2.2
 - files_downloadactivity: 1.1.1
 - files_markdown: 2.0.1
 - files_pdfviewer: 1.1.1
 - files_reader: 1.0.4
 - files_retention: 1.1.2
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - gpxedit: 0.0.7
 - gpxpod: 2.2.1
 - groupfolders: 1.1.0
 - issuetemplate: 0.3.0
 - keeweb: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - mail: 0.7.5
 - metadata: 0.6.0
 - nextcloud_announcements: 1.1
 - notes: 2.3.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - previewgenerator: 1.0.7
 - provisioning_api: 1.2.0
 - quota_warning: 1.1.1
 - ransomware_protection: 1.0.5
 - richdocuments: 1.12.38
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - spreed: 2.0.2
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0
Disabled:
 - encryption
 - files_external
 - user_external
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***"
    ],
    "datadirectory": "\/var\/ncdata\/nc_01\/data",
    "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "12.0.4.1",
    "dbname": "nc_cloud01",
    "dbhost": "localhost",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "UTC",
    "installed": true,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "\/var\/run\/redis\/redis.sock",
        "port": 0,
        "timeout": 0,
        "dbindex": 0
    },
    "filelocking.enabled": true,
    "appstore.experimental.enabled": "true",
    "updater.server.url": "https:\/\/updates.nextcloud.org\/updater_server\/",
    "updater.release.channel": "stable",
    "mail_smtpmode": "smtp",
    "mail_from_address": "testcloud",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "localhost",
    "mail_smtpport": "25",
    "enable_previews": true,
    "preview_max_x": 1024,
    "preview_max_y": 1024,
    "preview_max_scale_factor": 8,
    "trashbin_retention_obligation": "auto",
    "maintenance": false,
    "theme": "",
    "loglevel": 2,
    "updater.secret": "***REMOVED SENSITIVE VALUE***"
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

## Client configuration

**Browser:** Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko/20100101 Firefox/57.0

**Operating system:** Windows 7

## Logs

<details><summary>Browser log</summary>

```
Insert your webserver log here 
```
</details>

<details><summary>Nextcloud log</summary>

```
{"reqId":"Z4AvbjdSqByHEXD94Sm3","level":3,"time":"2017-11-29T14:48:23+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/accounts\/2\/folders\/SU5CT1g%3D\/messages?filter=","message":"Narrowing occurred during type inference. Please file a bug report on bugs.php.net at \/var\/www\/nc_01\/nextcloud\/lib\/composer\/composer\/ClassLoader.php#444","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"nHKOVJ1qqAYAekyXMMWO","level":3,"time":"2017-11-29T14:48:23+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_change_key_case() expects parameter 1 to be array, boolean given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/DataAccess.php#22","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"nHKOVJ1qqAYAekyXMMWO","level":3,"time":"2017-11-29T14:48:23+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#77","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"nHKOVJ1qqAYAekyXMMWO","level":3,"time":"2017-11-29T14:48:23+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"end() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#80","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"83iDj5mnccQs7LYUknbf","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_change_key_case() expects parameter 1 to be array, boolean given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/DataAccess.php#22","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"83iDj5mnccQs7LYUknbf","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#77","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"83iDj5mnccQs7LYUknbf","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"end() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#80","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"Ifih8viAVRVrzzkHo4Ac","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_change_key_case() expects parameter 1 to be array, boolean given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/DataAccess.php#22","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"Ifih8viAVRVrzzkHo4Ac","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#77","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"Ifih8viAVRVrzzkHo4Ac","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"end() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#80","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"7EH5FCHVnxR1bYlybjhH","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_change_key_case() expects parameter 1 to be array, boolean given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/DataAccess.php#22","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"7EH5FCHVnxR1bYlybjhH","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#77","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
{"reqId":"7EH5FCHVnxR1bYlybjhH","level":3,"time":"2017-11-29T14:48:24+00:00","remoteAddr":"24.134.13.209","user":"xxxxxxxx","app":"PHP","method":"GET","url":"\/index.php\/apps\/mail\/api\/avatars\/url\/xxxxxxxx","message":"end() expects parameter 1 to be array, null given at \/var\/www\/nc_01\/nextcloud\/apps\/mail\/vendor\/mpclarkson\/icon-scraper\/src\/IconScraper\/Scraper.php#80","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko\/20100101 Firefox\/57.0","version":"12.0.4.1"}
```
</details>

**Comment by RainerEmrich** [comment]
I removed the account names and E-Mail-Addresses from the log.

**Comment by ChristophWurst** [comment]
Fixed by  #651
